.github/workflows/gen-openapi.yml

```diff
@@ -0,0 +1,28 @@
+name: update-api-schema
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review, labeled, unlabeled]
+  push:
+    branches: [main]
```

Comment:
I think, we should not run this action on `main` directly. Only for pull request